Test.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+#include <stdexcept>
+
+class FactorialCalculator {
```

Comment:
Consider adding more detailed documentation and comments explaining the purpose of this class and its methods for better code readability and maintainability.